torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -808,13 +804,26 @@ void AliasDb::analyzeImpl(Node* node) {
       analysis == AliasAnalysisKind::FROM_SCHEMA,
       "AliasAnalysisKind::CONSERVATIVE/PURE_FUNCTION/INTERNAL_SPECIAL_CASE should already have been handled above");
   const auto& schema = node->schema();
-
+  torch::utils::SchemaInfo schema_info(schema);
+  if (!isFrozen_ && node->hasNamedInput("training")) {
+    auto value = constant_as<bool>(node->namedInput("training"));
+    schema_info.addArgumentValue("training", !value.has_value() || *value);
+  } else if (!isFrozen_ && node->hasNamedInput("use_input_stats")) {
+    auto value = constant_as<bool>(node->namedInput("use_input_stats"));
+    schema_info.addArgumentValue(
+        "use_input_stats", !value.has_value() || *value);
+  }
```

Comment:
actually do you think you could change the behavior of schema_info for this? I think this PR will technically work but on second thought I think schema_info isn't behaving correctly right now:

If we don't know the value of `training` (and we don't assign its value in schema_info) then I think we should conservatively assume that `batch_norm` is mutating inputs (in case training=True). But schemainfo right now will assume that training=False if we don't know its value, which means that if a user doesn't provide inputs then sometimes it will return incorrect mutation information.

And with this change I think this will need to be changed slightly to handle freezing differently